homeassistant/components/playstation_network/media_player.py

```diff
@@ -0,0 +1,139 @@
+    presence = {"availability": "", "lastAvailableDate": ""}
+    platform = {"status": "", "platform": ""}
+    title = {"name": "", "format": "", "imageURL": None, "playing": False}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Media Player Entity Setup."""
+    coordinator: PlaystationNetworkCoordinator = config_entry.runtime_data
+
+    if coordinator.data.platform.get("platform") is None:
+        username = coordinator.data.username
+        _LOGGER.warning(
+            "No console found associated with account: %s. -- Skipping creation of media player",
+            username,
+        )
+        return
+
+    async_add_entities([MediaPlayer(coordinator)])
+
+
+class MediaPlayer(PlaystationNetworkEntity, MediaPlayerEntity):
+    """Media player entity representing currently playing game."""
+
+    device_class = MediaPlayerDeviceClass.TV
+
+    def __init__(self, coordinator: PlaystationNetworkCoordinator) -> None:
+        """Initialize PSN MediaPlayer."""
+        super().__init__(coordinator)
+        self.psn: PlaystationNetworkData = self.coordinator.data
+        self._attr_has_entity_name = True
+
+    @property
+    def icon(self) -> str:
+        """Icon Getter."""
+        return "mdi:sony-playstation"
+
+    @property
+    def media_image_remotely_accessible(self) -> bool:
+        """Is media image remotely accessible getter."""
+        return True
+
+    @property
+    def state(self) -> MediaPlayerState:
+        """Media Player state getter."""
+        match self.psn.platform.get("onlineStatus", ""):
+            case "online":
+                if (
+                    self.psn.available is True
+                    and self.psn.title_metadata.get("npTitleId") is not None
+                ):
+                    return MediaPlayerState.PLAYING
+                return MediaPlayerState.ON
+            case "offline":
+                return MediaPlayerState.OFF
+            case _:
+                return MediaPlayerState.OFF
+
+    @property
+    def unique_id(self) -> str:
+        """Unique ID Getter."""
+        return f"{self.psn.username.lower()}_{self.psn.platform.get('platform', "").lower()}_console"
+
+    @property
+    def name(self) -> str:
+        """Name getter."""
+        return f"{self.psn.platform.get('platform')} Console"
+
+    @property
+    def media_content_type(self) -> MediaType:
+        """Content type of current playing media."""
+        return MediaType.GAME
+
+    @property
+    def media_title(self) -> str | None:
+        """Media title getter."""
+        if self.psn.title_metadata.get("npTitleId"):
+            return self.psn.title_metadata.get("titleName")
+        if self.psn.platform.get("onlineStatus") == "online":
+            return "Browsing the menu"
+        return None
+
+    @property
+    def app_name(self) -> str:
+        """App name getter."""
+        return ""
```

Comment:
Why an empty app_name?

Reply:
If I'm remembering right, the only thing that made sense to go here was the game name and it was already in the title.

Reply:
Oh, this is not for the game title, you put the game id here. On android tv for example it shows the id of the app like `com.netflix.mediaclient` for netflix.

Reply:
Right, but on an android tv, that information is actionable. You can use that id to switch the running app. But on the playstation the information is not very useful since we don't have control of the system. It also gunks up the media player control. Setting it to a blank string is wrong though. 

I started with just removing the app_name getter. If you really think the right thing to do is include the id, I'll add it in and we can bring it up during the pr review. 

![Screenshot 2024-11-26 at 7 52 42 AM](https://github.com/user-attachments/assets/dd0c3f81-acb9-4c68-96c1-f675f8d9e72f)

Reply:
Ah, I confused it with app_id 😬